include/rcutils/logging.h

```diff
@@ -516,17 +516,18 @@ void rcutils_logging_console_output_handler(
  * All logging macros ensure that this has been called once.
  */
 #define RCUTILS_LOGGING_AUTOINIT \
-  if (RCUTILS_UNLIKELY(!g_rcutils_logging_initialized)) { \
-    rcutils_ret_t ret = rcutils_logging_initialize(); \
-    if (ret != RCUTILS_RET_OK) { \
-      RCUTILS_SAFE_FWRITE_TO_STDERR( \
-        "[rcutils|" __FILE__ ":" RCUTILS_STRINGIFY(__LINE__) \
-        "] error initializing logging: "); \
-      RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string().str); \
-      RCUTILS_SAFE_FWRITE_TO_STDERR("\n"); \
-      rcutils_reset_error(); \
+  do { \
+    if (RCUTILS_UNLIKELY(!g_rcutils_logging_initialized)) { \
+      if (rcutils_logging_initialize() != RCUTILS_RET_OK) { \
+        RCUTILS_SAFE_FWRITE_TO_STDERR( \
+          "[rcutils|" __FILE__ ":" RCUTILS_STRINGIFY(__LINE__) \
+          "] error initializing logging: "); \
+        RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string().str); \
+        RCUTILS_SAFE_FWRITE_TO_STDERR("\n"); \
+        rcutils_reset_error(); \
+      } \
     } \
-  }
+  } while (0);
```

Comment:
This looks pretty good to me.  One thing that I'd prefer to do here is to remove the semicolon from the end of this line, and make the callers of this macro have the semicolon.  That will play more nicely with text editors, and also look more like statements inline.

To do this, we'd have to update all of the callers of `RCUTILS_LOGGING_AUTOINIT` to have a semicolon on the end.  Most of the callers are in this repository, but there is also a caller in [rcl](https://github.com/ros2/rcl/blob/f3396a2a7dcf061d92642a8dd2af0b1673d36057/rcl/src/rcl/logging.c#L65) and [rclpy](https://github.com/ros2/rclpy/blob/a37c01511324c1e3e57bc47a7e1286bf79514c60/rclpy/src/rclpy/_rclpy_logging.c#L173).  We'd need to have separate PRs for that.

@felixendres Are you willing to make those updates?  If so, please go ahead and fix up rcutils as suggested, and open PRs to rcl and rclpy as well.  If not, that's OK, but please open an issue so we can do this follow-up in the future.

Reply:
I totally agree on @clalancette suggestion, in addition to

- No need to prevent uncrustify from making unnecessary indents.
- It is unlikely that user application calls `RCUTILS_LOGGING_AUTOINIT` directly.